cmd/tier.go

```diff
@@ -501,6 +512,27 @@ func NewTierConfigMgr() *TierConfigMgr {
 	}
 }
 
+func (config *TierConfigMgr) refreshTierConfig(ctx context.Context, objAPI ObjectLayer) {
+	const tierCfgRefresh = 15 * time.Minute
+	t := time.NewTimer(tierCfgRefresh)
+	sleeper := newDynamicSleeper(2, 150*time.Millisecond, false)
+	defer t.Stop()
+	for {
+		wait := sleeper.Timer(ctx)
+		select {
+		case <-ctx.Done():
+			return
+		case <-t.C:
+			err := config.Reload(ctx, objAPI)
+			if err != nil {
+				logger.LogIf(ctx, err)
+			}
+			wait()
```

Comment:
the logic of this wait is a bit weird, it is not that we are not going to sleep here, we still reset the timer here, therefore we are going to sleep for 15 minutes

Reply:
I think it's a wait() after the config has hit the disk. Basically an additional time to wait for the servers, and let everyone not read at the same time.

Reply:
For that to happen we need it the sleep before Reload() not after.

Reply:
@vadmeste good catch. @harshavardhana yes, that was my intention. I now realize that a random sleep of up to 10s is the correct approach for avoiding all MinIO servers contending for the read-locks required to fetch the tier configuration object.
Let me address this.

Reply:
if you want to stagger it tagger it like bucketMetadataSys reload() we have to stagger it before the read()

Reply:
PTAL @vadmeste @harshavardhana.